dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -427,4 +464,23 @@ static String enclosingClassName(String className) {
     }
     return className.substring(0, lastDollarIndex);
   }
+
+  /**
+   * Returns the target class and its superclasses in order (with {@link Object} last). If any can't
+   * be found, the list stops with the previous one.
+   */
+  static Iterable<JavaClass> getClassAndSuperClasses(JavaClass targetClass) {
+    return SUPERCLASSES.breadthFirst(targetClass);
```

Comment:
breadthFirst sounds strange here. We're just walking up a tree, right?

Reply:
It's correct although the method name may sound strange when Guava's Traverser is used for class hierarchy.

Guava's Traverser (`SUPERCLASSES` below) defines graph's (directed) edges. In this case, it defines the relationship from a class to its parent class. Because a Java class has only one parent class, there's only one (outgoing) edge from a node in the graph defined by `SUPERCLASSES`. So `breadthFirst` just works (so does `depthFirst`).

Reply:
Makes me wonder if Traverser is overkill here.